iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/helplets.inl

```diff
@@ -24,29 +24,30 @@ namespace cxx
 {
 template <uint64_t StringCapacity>
 inline bool isValidPathEntry(const string<StringCapacity>& name,
-                             const RelativePathComponents& relativePathComponents) noexcept
+                             const RelativePathComponents relativePathComponents) noexcept
 {
-    const string<StringCapacity> currentDirectory(".");
-    const string<StringCapacity> parentDirectory("..");
+    const string<StringCapacity> currentDirectory{"."};
+    const string<StringCapacity> parentDirectory{".."};
 
-    if (name == currentDirectory || name == parentDirectory)
+    if ((name == currentDirectory) || (name == parentDirectory))
     {
         return relativePathComponents == RelativePathComponents::ACCEPT;
     }
 
     const auto nameSize = name.size();
 
-    for (uint64_t i = 0; i < nameSize; ++i)
+    for (uint64_t i{0}; i < nameSize; ++i)
     {
-        const char c = name[i];
+        // AXIVION Next Construct AutosarC++19_03-A3.9.1: Basic numerical type of char shall be used
+        const char c{name[i]};
 
-        const bool isSmallLetter = (internal::ASCII_A <= c && c <= internal::ASCII_Z);
-        const bool isCapitalLetter = (internal::ASCII_CAPITAL_A <= c && c <= internal::ASCII_CAPITAL_Z);
-        const bool isNumber = (internal::ASCII_0 <= c && c <= internal::ASCII_9);
-        const bool isSpecialCharacter = (c == internal::ASCII_MINUS || c == internal::ASCII_DOT
-                                         || c == internal::ASCII_COLON || c == internal::ASCII_UNDERSCORE);
+        const bool isSmallLetter{((internal::ASCII_A <= c) && (c <= internal::ASCII_Z))};
```

Comment:
We should avoid that many parentheses if possible. I disagree with the rule in general, but even with the rule

```
const bool isSmallLetter{(internal::ASCII_A <= c) && (c <= internal::ASCII_Z)};
```
should suffice.

Similar for the other occurrences. The outer parentheses at the very least should be removed.